From c3651bf4763d7247e3edd4e20526a85de459041b Mon Sep 17 00:00:00 2001
From: Srivats P <srivats.p@conexant.com>
Date: Fri, 27 Mar 2009 11:17:26 -0700
Subject: [PATCH 08/23] ip6tunnel: Fix no default display of ip4ip6 tunnels

"ip -6 tunnel show" displays only ip6ip6 tunnels not ip4ip6 tunnels
 - it should display all irrespective of proto.

This is because the default tunnel proto is initialized to IPPROTO_IPV6 in ip6_tnl_parm_init() which is fine for a 'add' command but not for 'show'. This patch overrides proto with 0 signifying 'mode any' as the default in case of a 'show'.
---
 ip/ip6tunnel.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/ip/ip6tunnel.c b/ip/ip6tunnel.c
index 8421983..8852a67 100644
--- a/ip/ip6tunnel.c
+++ b/ip/ip6tunnel.c
@@ -335,6 +335,7 @@ static int do_show(int argc, char **argv)
         struct ip6_tnl_parm p;
 
 	ip6_tnl_parm_init(&p, 0);
+	p.proto = 0;  /* default to any */
 
         if (parse_args(argc, argv, &p) < 0)
                 return -1;
-- 
1.6.3.3

